functional test (retrieval) exits prematurely when configured for large sectors

### Description

The script exits after the `wait for commitSector sent by miner owner to be included in a block viewable by both nodes...` message is written to stdout.

```
Building go-filecoin...
git log -n 1 --format=%H
go build -ldflags -X github.com/filecoin-project/go-filecoin/flags.Commit=59a6167d1b6c5ade0865eac60d771928714ad6f5 -v -o go-filecoin .
github.com/filecoin-project/go-filecoin/node
Building gengen utils...
go build -o ./gengen/gengen ./gengen
Building faucet...
go build -o ./tools/faucet/faucet ./tools/faucet/
Building genesis file server...
go build -o ./tools/genesis-file-server/genesis-file-server ./tools/genesis-file-server/
Generating genesis...
./gengen/gengen --keypath fixtures --out-car fixtures/genesis.car --out-json fixtures/gen.json --config ./fixtures/setup.json
1040000+0 records in
1040000+0 records out
1064960000 bytes transferred in 25.700942 secs (41436613 bytes/sec)
500+0 records in
500+0 records out
512000 bytes transferred in 0.011303 secs (45297917 bytes/sec)

generating private keys...

initializing daemons...
initializing filecoin node at /var/folders/8h/gmmbhq1s5tb8sk9011sy2wkw0000gn/T/tmp.IrfMFlTi
initializing filecoin node at /var/folders/8h/gmmbhq1s5tb8sk9011sy2wkw0000gn/T/tmp.3H4WoDbA

start daemons...
My peer ID is QmQ1DQmyb4Tw6sMaakgCZ9EBbNRmtfKxiTDEnALMWaqzf5
My peer ID is QmQx56H2tYhqqsXvpvrpBMgiCL61M6bmUdKCEtogcqb7ZR
Swarm listening on: /ip4/127.0.0.1/tcp/60016
Swarm listening on: /ip4/127.0.0.1/tcp/60018

importing private keys...

ensure that miner address is set so that the miner-node can mine...
minerAddress = "fcq8eut7ekjvkzt0dfxhrv4wzhhgcm34crfdlnw5j"

node default address should match what's associated with imported SK...
defaultAddress = "fcqarhhu6kqrptk2u5dwu4nn8990ca6y0km5z50j2"
defaultAddress = "fcqwtrjf45rljynf2e8fw8klewy34k63tgymxj9c6"

get mining node's libp2p identity...

get the client's peer id...

update miner's libp2p identity to match its node's...

connecting daemons...
connect QmQx56H2tYhqqsXvpvrpBMgiCL61M6bmUdKCEtogcqb7ZR success
connect QmQ1DQmyb4Tw6sMaakgCZ9EBbNRmtfKxiTDEnALMWaqzf5 success



********************** BEGIN STORAGE PROTOCOL




miner mines a block (to include commitSector message)...
zDPWYqFD8u6AmsZmwTaZ9WgaRsV1cMsxRyoRPPL1WkThScPMA7jv

miner node starts mining...
Started mining

use process substitution, waiting until messages are in blockchain...

block until miner peer id-update message appears in chains...
waiting for: zDPWYqFCsKe2n7tYW4vrDUM1gKd12k4WE8TVJFpjTZ1gegAYNJk5
waiting for: zDPWYqFD5SyBKLnPvGwSJwYfsAm7vcxb6yWuSB4FWZocLNgi3wdp
waiting for: zDPWYqFCsKe2n7tYW4vrDUM1gKd12k4WE8TVJFpjTZ1gegAYNJk5
waiting for: zDPWYqFD5SyBKLnPvGwSJwYfsAm7vcxb6yWuSB4FWZocLNgi3wdp
{ "message": { "to": "fcq8eut7ekjvkzt0dfxhrv4wzhhgcm34crfdlnw5j", "from": "fcqwtrjf45rljynf2e8fw8klewy34k63tgymxj9c6", "nonce": "Ag==", "value": null, "method": "updatePeerID", "params": "gVgiEiAYvLuoQ/loHdQPw9fE2xWSVqifYocrAJcb039mmEIy1A==" }, "signature": "c1iccHH8HCJuzYYbQpxy/zIOAA1gayFe0EhMJR3aqO8CuCjsx8HKemaeN4fPmU1u3c9pUvP0CYEfNNekBmRCbgE=" } { "exitCode": 0, "return": null }
{ "message": { "to": "fcq8eut7ekjvkzt0dfxhrv4wzhhgcm34crfdlnw5j", "from": "fcqwtrjf45rljynf2e8fw8klewy34k63tgymxj9c6", "nonce": "Ag==", "value": null, "method": "updatePeerID", "params": "gVgiEiAYvLuoQ/loHdQPw9fE2xWSVqifYocrAJcb039mmEIy1A==" }, "signature": "c1iccHH8HCJuzYYbQpxy/zIOAA1gayFe0EhMJR3aqO8CuCjsx8HKemaeN4fPmU1u3c9pUvP0CYEfNNekBmRCbgE=" } { "exitCode": 0, "return": null }
{ "message": { "to": "fcq8eut7ekjvkzt0dfxhrv4wzhhgcm34crfdlnw5j", "from": "fcqwtrjf45rljynf2e8fw8klewy34k63tgymxj9c6", "nonce": "Aw==", "value": null, "method": "addAsk", "params": "gkqAgKDPyODI44oBQicQ" }, "signature": "OCpAf4HMjYcYH6odA4IkTVe0GCg5g2i23DcArkKgkx8J/F5+T/h0AUgMn7PaavFBr18IpZ022b/ILJxwZVhxywE=" } { "exitCode": 0, "return": [ "" ] }
{ "message": { "to": "fcq8eut7ekjvkzt0dfxhrv4wzhhgcm34crfdlnw5j", "from": "fcqwtrjf45rljynf2e8fw8klewy34k63tgymxj9c6", "nonce": "Aw==", "value": null, "method": "addAsk", "params": "gkqAgKDPyODI44oBQicQ" }, "signature": "OCpAf4HMjYcYH6odA4IkTVe0GCg5g2i23DcArkKgkx8J/F5+T/h0AUgMn7PaavFBr18IpZ022b/ILJxwZVhxywE=" } { "exitCode": 0, "return": [ "" ] }

client imports piece 1...

client proposes a storage deal, which transfers file 1...
State:   accepted
Message:
DealID:  zdpuAyA2tMACSupHB5GrpWzjXMos9tc13T5TR39mU3y9c1nx2

client imports piece 2...

client proposes a storage deal, which transfers piece 2 (triggers seal)...
State:   accepted
Message:
DealID:  zdpuAmqZBJpPrPqEM3JedbozjvT5fZqZfWh4xTWb2m2NDiDW3

miner mines a block (to include commitSector message)...
zDPWYqFD1t7owvdFJt9sTZhKfuQt7WhUbS7VCmi2tRcobZCQuVcM

wait for commitSector sent by miner owner to be included in a block viewable by both nodes...

cleaning up...
Got terminated, shutting down...
Got terminated, shutting down...
07:40:10.236 ERROR       node: MessageSub.Next(): subscription cancelled by calling sub.Cancel() node.go:1158
07:40:10.236 ERROR       node: BlockSub.Next(): context canceled node.go:1158
07:40:10.236 ERROR       node: BlockSub.Next(): subscription cancelled by calling sub.Cancel() node.go:1158
07:40:10.236 ERROR       node: MessageSub.Next(): subscription cancelled by calling sub.Cancel() node.go:1158
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x4551c61]

goroutine 348 [running]:
gx/ipfs/Qmb3GBFCHMuzmi9EpH3pxpYBiviyc3tEPyDQHxZJQJSxj9/badger/skl.(*Skiplist).IncrRef(...)
	/Users/erinswenson-healey/go/src/gx/ipfs/Qmb3GBFCHMuzmi9EpH3pxpYBiviyc3tEPyDQHxZJQJSxj9/badger/skl/skl.go:86
gx/ipfs/Qmb3GBFCHMuzmi9EpH3pxpYBiviyc3tEPyDQHxZJQJSxj9/badger.(*DB).getMemTables(0xc000032e00, 0x0, 0x0, 0x0, 0x0)
	/Users/erinswenson-healey/go/src/gx/ipfs/Qmb3GBFCHMuzmi9EpH3pxpYBiviyc3tEPyDQHxZJQJSxj9/badger/db.go:472 +0xd1
gx/ipfs/Qmb3GBFCHMuzmi9EpH3pxpYBiviyc3tEPyDQHxZJQJSxj9/badger.(*Txn).NewIterator(0xc016fb2bd0, 0x1, 0x64, 0x0, 0x0)
	/Users/erinswenson-healey/go/src/gx/ipfs/Qmb3GBFCHMuzmi9EpH3pxpYBiviyc3tEPyDQHxZJQJSxj9/badger/iterator.go:317 +0x50
gx/ipfs/QmciWFTxK1t1PNyPdoTsv8wtj3DWim2qRFjaUEcFWCnV8D/go-ds-badger.(*Datastore).QueryNew(0xc0000af190, 0xc016606550, 0x42, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/Users/erinswenson-healey/go/src/gx/ipfs/QmciWFTxK1t1PNyPdoTsv8wtj3DWim2qRFjaUEcFWCnV8D/go-ds-badger/datastore.go:143 +0xf0
gx/ipfs/QmciWFTxK1t1PNyPdoTsv8wtj3DWim2qRFjaUEcFWCnV8D/go-ds-badger.(*Datastore).Query(0xc0000af190, 0xc016606550, 0x42, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/Users/erinswenson-healey/go/src/gx/ipfs/QmciWFTxK1t1PNyPdoTsv8wtj3DWim2qRFjaUEcFWCnV8D/go-ds-badger/datastore.go:133 +0x6d
gx/ipfs/QmVG5gxteQNEMhrS8prJSmU2C9rebtFuTd3SYZ5kE3YZ5k/go-datastore/autobatch.(*Datastore).Query(0xc0004abf60, 0xc016606550, 0x42, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/Users/erinswenson-healey/go/src/gx/ipfs/QmVG5gxteQNEMhrS8prJSmU2C9rebtFuTd3SYZ5kE3YZ5k/go-datastore/autobatch/autobatch.go:93 +0xc0
gx/ipfs/Qmb8TxXJEnL65XnmkEZfGd8mEFU6cxptEP4oCfTvcDusd8/go-libp2p-kad-dht/providers.loadProvSet(0x4b40ea0, 0xc0004abf60, 0xc01a42d260, 0x0, 0x0, 0xc0168f3f00)
	/Users/erinswenson-healey/go/src/gx/ipfs/Qmb8TxXJEnL65XnmkEZfGd8mEFU6cxptEP4oCfTvcDusd8/go-libp2p-kad-dht/providers/providers.go:115 +0xac
gx/ipfs/Qmb8TxXJEnL65XnmkEZfGd8mEFU6cxptEP4oCfTvcDusd8/go-libp2p-kad-dht/providers.(*ProviderManager).addProv(0xc0004af680, 0xc01a42d260, 0xc0000b66c0, 0x22, 0x1, 0x0)
	/Users/erinswenson-healey/go/src/gx/ipfs/Qmb8TxXJEnL65XnmkEZfGd8mEFU6cxptEP4oCfTvcDusd8/go-libp2p-kad-dht/providers/providers.go:167 +0x2fe
gx/ipfs/Qmb8TxXJEnL65XnmkEZfGd8mEFU6cxptEP4oCfTvcDusd8/go-libp2p-kad-dht/providers.(*ProviderManager).run(0xc0004af680)
	/Users/erinswenson-healey/go/src/gx/ipfs/Qmb8TxXJEnL65XnmkEZfGd8mEFU6cxptEP4oCfTvcDusd8/go-libp2p-kad-dht/providers/providers.go:256 +0xa9f
gx/ipfs/Qmb8TxXJEnL65XnmkEZfGd8mEFU6cxptEP4oCfTvcDusd8/go-libp2p-kad-dht/providers.NewProviderManager.func1(0x4b45460, 0xc0004af740)
	/Users/erinswenson-healey/go/src/gx/ipfs/Qmb8TxXJEnL65XnmkEZfGd8mEFU6cxptEP4oCfTvcDusd8/go-libp2p-kad-dht/providers/providers.go:73 +0x2a
gx/ipfs/QmSF8fPo3jgVBAy8fpdjjYqgG87dkJgUprRBHRd2tmfgpP/goprocess.(*process).Go.func1(0xc0004b0ad0, 0xc0004af740, 0xc0004af7a0)
	/Users/erinswenson-healey/go/src/gx/ipfs/QmSF8fPo3jgVBAy8fpdjjYqgG87dkJgUprRBHRd2tmfgpP/goprocess/impl-mutex.go:112 +0x3c
created by gx/ipfs/QmSF8fPo3jgVBAy8fpdjjYqgG87dkJgUprRBHRd2tmfgpP/goprocess.(*process).Go
	/Users/erinswenson-healey/go/src/gx/ipfs/QmSF8fPo3jgVBAy8fpdjjYqgG87dkJgUprRBHRd2tmfgpP/goprocess/impl-mutex.go:111 +0x1f5
stopping filecoin :(
```

### Where to Begin

If the following two lines are removed from the script, the script does not exit prematurely:

```
set -e
set -o pipefail
```

Note: Those two lines did not exist in the original MEGASCRIPT.

--- a/functional-tests/retrieval.sh
+++ b/functional-tests/retrieval.sh
@@ -8,18 +8,18 @@ set -o pipefail
 export GO_FILECOIN_LOG_LEVEL=3
 export FILECOIN_PROOFS_FAST_DELAY_SECONDS=1
 
-if [ -z $1 ]; then
+if [ -z "$1" ]; then
     USE_SMALL_SECTORS="true"
 else
-    USE_SMALL_SECTORS=$1
+    USE_SMALL_SECTORS="$1"
 fi
 
 export FIL_USE_SMALL_SECTORS=${USE_SMALL_SECTORS}
 
-if [ -z $2 ]; then
+if [ -z "$2" ]; then
     AUTO_SEAL_INTERVAL_SECONDS="0"
 else
-    AUTO_SEAL_INTERVAL_SECONDS=$2
+    AUTO_SEAL_INTERVAL_SECONDS="$2"
 fi
 
 # forward-declare stuff that we need to clean up
@@ -34,15 +34,15 @@ BLOCK_TIME="5s"
 HODL="HODL HODL HODL HODL HODL HODL HODL HODL HODL HODL HODL HODL HODL HODL HODL"
 
 if [ "${USE_SMALL_SECTORS}" = true ] ; then
-    echo ${HODL} > ${PIECE_1_PATH}
-    echo "BLOCKCHAIN BLOCKCHAIN BLOCKCHAIN BLOCKCHAIN BLOCKCHAIN BLOCKCHAIN BLOCKCHAIN" > ${PIECE_2_PATH}
+    echo "${HODL}" > "${PIECE_1_PATH}"
+    echo "BLOCKCHAIN BLOCKCHAIN BLOCKCHAIN BLOCKCHAIN BLOCKCHAIN BLOCKCHAIN BLOCKCHAIN" > "${PIECE_2_PATH}"
 else
     # Keeping first piece block count at: 1040000 < 1040384 = 1065353216/1024
     # will prevent first piece from triggering seal.
-    dd if=/dev/urandom of=${PIECE_1_PATH} bs=1024 count=1040000
+    dd if=/dev/urandom of="${PIECE_2_PATH}" bs=1024 count=1040000
     # A total block count of: (piece_1_count + piece_2_count) = 1040000 + 500 = 1040500 > 1040384
     # will cause the second piece to trigger a seal.
-    dd if=/dev/urandom of=${PIECE_2_PATH} bs=1024 count=500
+    dd if=/dev/urandom of="${PIECE_1_PATH}" bs=1024 count=500
 fi
 
 function finish {
@@ -50,11 +50,11 @@ function finish {
   echo "cleaning up..."
   kill $MN_PID || true
   kill $CL_PID || true
-  rm -f ${PIECE_1_PATH}
-  rm -f ${PIECE_2_PATH}
-  rm -f ${UNSEAL_PATH}
-  rm -rf ${CL_REPO_DIR}
-  rm -rf ${MN_REPO_DIR}
+  rm -f "${PIECE_1_PATH}"
+  rm -f "${PIECE_2_PATH}"
+  rm -f "${UNSEAL_PATH}"
+  rm -rf "${CL_REPO_DIR}"
+  rm -rf "${MN_REPO_DIR}"
 }
 
 trap finish EXIT
@@ -64,61 +64,62 @@ function free_port {
 }
 
 function import_private_key {
-  ./go-filecoin wallet import ./fixtures/$1.key \
-    --repodir=$2 \
+  ./go-filecoin wallet import ./fixtures/"$1".key \
+    --repodir="$2" \
     | jq -r ""
 }
 
 function init_daemon {
   ./go-filecoin init \
-    --auto-seal-interval-seconds=${AUTO_SEAL_INTERVAL_SECONDS} \
-    --repodir=$1 \
-    --cmdapiaddr=/ip4/127.0.0.1/tcp/$2 \
+    --auto-seal-interval-seconds="${AUTO_SEAL_INTERVAL_SECONDS}" \
+    --repodir="$1" \
+    --cmdapiaddr=/ip4/127.0.0.1/tcp/"$2" \
     --walletfile= \
     --walletaddr= \
     --testgenesis=false \
-    --genesisfile=$3
+    --genesisfile="$3"
 }
 
 function start_daemon {
   ./go-filecoin daemon \
-    --repodir=$1 \
-    --block-time=${BLOCK_TIME} \
-    --cmdapiaddr=/ip4/127.0.0.1/tcp/$2 \
-    --swarmlisten=/ip4/127.0.0.1/tcp/$3 &
+    --repodir="$1" \
+    --block-time="${BLOCK_TIME}" \
+    --cmdapiaddr=/ip4/127.0.0.1/tcp/"$2" \
+    --swarmlisten=/ip4/127.0.0.1/tcp/"$3" &
 }
 
 function get_first_address {
   ./go-filecoin id \
-    --repodir=$1 \
+    --repodir="$1" \
     | jq -r ".Addresses[0]"
 }
 
 function get_peer_id {
   ./go-filecoin id \
-    --repodir=$1 \
+    --repodir="$1" \
     | jq -r ".ID"
 }
 
 function get_peers {
   ./go-filecoin swarm peers \
-    --repodir=$1
+    --repodir="$1"
 }
 
 function swarm_connect {
-  ./go-filecoin swarm connect $1 \
-    --repodir=$2
+  ./go-filecoin swarm connect "$1" \
+    --repodir="$2"
+    local __peers
 
-  local __peers=$(get_peers $2)
+  __peers=$(get_peers "$2")
   until [[ "$__peers" = "$1" ]]; do
-    __peers=$(get_peers $2)
+    __peers=$(get_peers "$2")
     sleep 1
   done
 }
 
 function chain_ls {
   ./go-filecoin chain ls --enc=json \
-    --repodir=$1
+    --repodir="$1"
 }
 
 function wait_for_message_in_chain_by_method_and_sender {
@@ -128,18 +129,19 @@ function wait_for_message_in_chain_by_method_and_sender {
   local __hodl=""
 
   while [ -z $__hodl ]; do
-    __chain=$(chain_ls $3)
+    __chain=$(chain_ls "$3")
 
     __hodl=""
     for blk in $__chain
     do
-        __hodl=$(echo "$blk" | jq ".[].messages[].message | select(.method == \"$1\").from | select(. == \"$2\")" 2>/dev/null | head -n 1)
-        if [ ! -z $__hodl ]; then
+        __hodl=$(echo "$blk" | jq ".[].messages[].message | select(.method == \"$1\").from | select(. == \"$2\")" 2>/dev/null | head -n 1 || true)
+        if [ ! -z "$__hodl" ]; then
           break
         fi
     done
 
     echo "$(date "+%T") - sleeping for 10 seconds"
+    echo "$__hodl"
     sleep 10
   done
 
@@ -147,34 +149,34 @@ function wait_for_message_in_chain_by_method_and_sender {
 }
 
 function set_wallet_default_address_in_config {
-  ./go-filecoin config wallet.defaultAddress \"$1\" \
-    --repodir=$2
+  ./go-filecoin config wallet.defaultAddress \""$1"\" \
+    --repodir="$2"
 }
 
 function set_mining_address_in_config {
-  ./go-filecoin config mining.minerAddress \"$1\" \
-    --repodir=$2
+  ./go-filecoin config mining.minerAddress \""$1"\" \
+    --repodir="$2"
 }
 
 function wait_mpool_size {
   ./go-filecoin mpool \
-    --wait-for-count=$1 \
+    --wait-for-count="$1" \
     --repodir="$2"
 }
 
 function add_ask {
-  ./go-filecoin miner add-ask $1 $2 $3 \
+  ./go-filecoin miner add-ask "$1" "$2" "$3" \
     --repodir="$4"
 }
 
 function miner_update_pid {
-  ./go-filecoin miner update-peerid $1 $2 \
-    --repodir=$3
+  ./go-filecoin miner update-peerid "$1" "$2" \
+    --repodir="$3"
 }
 
 function mine_once {
   ./go-filecoin mining once \
-    --repodir=$1
+    --repodir="$1"
 }
 
 function fork_message_wait {
@@ -182,7 +184,7 @@ function fork_message_wait {
 }
 
 function join {
-  echo $(eval "cat <&$1")
+  cat <&"$1"
 }
 
 MN_REPO_DIR=$(mktemp -d)
@@ -196,53 +198,53 @@ CL_SWARM_PORT=$(free_port)
 echo ""
 echo "generating private keys..."
 
-MN_MINER_FIL_ADDR=$(cat fixtures/gen.json | jq -r '.Miners[] | select(.Owner == 0).Address')
+MN_MINER_FIL_ADDR=$(jq -r '.Miners[] | select(.Owner == 0).Address' < fixtures/gen.json)
 
 echo ""
 echo "initializing daemons..."
-init_daemon ${MN_REPO_DIR} ${MN_CMDAPI_PORT} ./fixtures/genesis.car
-init_daemon ${CL_REPO_DIR} ${CL_CMDAPI_PORT} ./fixtures/genesis.car
+init_daemon "${MN_REPO_DIR}" "${MN_CMDAPI_PORT}" ./fixtures/genesis.car
+init_daemon "${CL_REPO_DIR}" "${CL_CMDAPI_PORT}" ./fixtures/genesis.car
 
 echo ""
 echo "start daemons..."
-start_daemon ${MN_REPO_DIR} ${MN_CMDAPI_PORT} ${MN_SWARM_PORT}
+start_daemon "${MN_REPO_DIR}" "${MN_CMDAPI_PORT}" "${MN_SWARM_PORT}"
 MN_PID=$!
-start_daemon ${CL_REPO_DIR} ${CL_CMDAPI_PORT} ${CL_SWARM_PORT}
+start_daemon "${CL_REPO_DIR}" "${CL_CMDAPI_PORT}" "${CL_SWARM_PORT}"
 CL_PID=$!
 
 sleep 2
 
 echo ""
 echo "importing private keys..."
-MN_MINER_OWNER_FIL_ADDR=$(import_private_key 0 ${MN_REPO_DIR})
-CL_FIL_ADDRESS=$(import_private_key 1 ${CL_REPO_DIR})
+MN_MINER_OWNER_FIL_ADDR=$(import_private_key 0 "${MN_REPO_DIR}")
+CL_FIL_ADDRESS=$(import_private_key 1 "${CL_REPO_DIR}")
 
 echo ""
 echo "ensure that miner address is set so that the miner-node can mine..."
-set_mining_address_in_config ${MN_MINER_FIL_ADDR} ${MN_REPO_DIR}
+set_mining_address_in_config "${MN_MINER_FIL_ADDR}" "${MN_REPO_DIR}"
 
 echo ""
 echo "node default address should match what's associated with imported SK..."
-set_wallet_default_address_in_config ${CL_FIL_ADDRESS} ${CL_REPO_DIR}
-set_wallet_default_address_in_config ${MN_MINER_OWNER_FIL_ADDR} ${MN_REPO_DIR}
+set_wallet_default_address_in_config "${CL_FIL_ADDRESS}" "${CL_REPO_DIR}"
+set_wallet_default_address_in_config "${MN_MINER_OWNER_FIL_ADDR}" "${MN_REPO_DIR}"
 
 echo ""
 echo "get mining node's libp2p identity..."
-MN_PEER_ID=$(get_peer_id ${MN_REPO_DIR})
+MN_PEER_ID=$(get_peer_id "${MN_REPO_DIR}")
 
-echo ""
-echo "get the client's peer id..."
-CL_PEER_ID=$(get_peer_id ${CL_REPO_DIR})
+#echo ""
+#echo "get the client's peer id..."
+#CL_PEER_ID=$(get_peer_id "${CL_REPO_DIR}")
 
 echo ""
 echo "update miner's libp2p identity to match its node's..."
-MINER_UPDATE_PID_MSG_CID=$(miner_update_pid ${MN_MINER_FIL_ADDR} ${MN_PEER_ID} ${MN_REPO_DIR})
-MINER_ADD_ASK_MSG_CID=$(add_ask ${MN_MINER_FIL_ADDR} 10 10000 ${MN_REPO_DIR})
+MINER_UPDATE_PID_MSG_CID=$(miner_update_pid "${MN_MINER_FIL_ADDR}" "${MN_PEER_ID}" "${MN_REPO_DIR}")
+MINER_ADD_ASK_MSG_CID=$(add_ask "${MN_MINER_FIL_ADDR}" 10 10000 "${MN_REPO_DIR}")
 
 echo ""
 echo "connecting daemons..."
-swarm_connect $(get_first_address ${CL_REPO_DIR}) ${MN_REPO_DIR}
-swarm_connect $(get_first_address ${MN_REPO_DIR}) ${CL_REPO_DIR}
+swarm_connect "$(get_first_address "${CL_REPO_DIR}")" "${MN_REPO_DIR}"
+swarm_connect "$(get_first_address "${MN_REPO_DIR}")" "${CL_REPO_DIR}"
 
 echo ""
 echo ""
@@ -254,19 +256,19 @@ echo ""
 
 echo ""
 echo "miner mines a block (to include commitSector message)..."
-mine_once ${MN_REPO_DIR}
+mine_once "${MN_REPO_DIR}"
 
 echo ""
 echo "miner node starts mining..."
 ./go-filecoin mining start \
-  --repodir=$MN_REPO_DIR \
+  --repodir="$MN_REPO_DIR" \
 
 echo ""
 echo "use process substitution, waiting until messages are in blockchain..."
-fork_message_wait 104 ${MINER_UPDATE_PID_MSG_CID} ${MN_REPO_DIR}
-fork_message_wait 105 ${MINER_UPDATE_PID_MSG_CID} ${CL_REPO_DIR}
-fork_message_wait 106 ${MINER_ADD_ASK_MSG_CID} ${MN_REPO_DIR}
-fork_message_wait 107 ${MINER_ADD_ASK_MSG_CID} ${CL_REPO_DIR}
+fork_message_wait 104 "${MINER_UPDATE_PID_MSG_CID}" "${MN_REPO_DIR}"
+fork_message_wait 105 "${MINER_UPDATE_PID_MSG_CID}" "${CL_REPO_DIR}"
+fork_message_wait 106 "${MINER_ADD_ASK_MSG_CID}" "${MN_REPO_DIR}"
+fork_message_wait 107 "${MINER_ADD_ASK_MSG_CID}" "${CL_REPO_DIR}"
 
 echo ""
 echo "block until miner peer id-update message appears in chains..."
@@ -277,33 +279,31 @@ join 107
 
 echo ""
 echo "client imports piece 1..."
-PIECE_1_CID=$(cat ${PIECE_1_PATH} \
-  | ./go-filecoin client import --repodir=${CL_REPO_DIR})
+PIECE_1_CID=$(./go-filecoin client import --repodir="${CL_REPO_DIR}" < "${PIECE_1_PATH}")
 
 echo ""
 echo "client proposes a storage deal, which transfers file 1..."
-./go-filecoin client propose-storage-deal ${MN_MINER_FIL_ADDR} ${PIECE_1_CID} 0 5 \
-  --repodir=$CL_REPO_DIR \
+./go-filecoin client propose-storage-deal "${MN_MINER_FIL_ADDR}" "${PIECE_1_CID}" 0 5 \
+  --repodir="$CL_REPO_DIR" \
 
 echo ""
 echo "client imports piece 2..."
-PIECE_2_CID=$(cat ${PIECE_2_PATH} \
-  | ./go-filecoin client import --repodir=${CL_REPO_DIR})
+PIECE_2_CID=$(./go-filecoin client import --repodir="${CL_REPO_DIR}" < "${PIECE_2_PATH}")
 
 echo ""
 echo "client proposes a storage deal, which transfers piece 2 (triggers seal)..."
-./go-filecoin client propose-storage-deal ${MN_MINER_FIL_ADDR} ${PIECE_2_CID} 0 5 \
-  --repodir=$CL_REPO_DIR \
+./go-filecoin client propose-storage-deal "${MN_MINER_FIL_ADDR}" "${PIECE_2_CID}" 0 5 \
+  --repodir="$CL_REPO_DIR" \
 
 echo ""
 echo "miner mines a block (to include commitSector message)..."
-mine_once ${MN_REPO_DIR}
+mine_once "${MN_REPO_DIR}"
 
 echo ""
 echo "wait for commitSector sent by miner owner to be included in a block viewable by both nodes..."
-wait_for_message_in_chain_by_method_and_sender commitSector ${MN_MINER_OWNER_FIL_ADDR} ${CL_REPO_DIR}
+wait_for_message_in_chain_by_method_and_sender commitSector "${MN_MINER_OWNER_FIL_ADDR}" "${CL_REPO_DIR}"
 
-wait_for_message_in_chain_by_method_and_sender commitSector ${MN_MINER_OWNER_FIL_ADDR} ${MN_REPO_DIR}
+wait_for_message_in_chain_by_method_and_sender commitSector "${MN_MINER_OWNER_FIL_ADDR}" "${MN_REPO_DIR}"
 echo ""
 echo ""
 echo ""
@@ -312,27 +312,21 @@ echo ""
 echo ""
 echo ""
 
-./go-filecoin retrieval-client retrieve-piece ${MN_PEER_ID} ${PIECE_1_CID} \
-  --repodir=${CL_REPO_DIR} > ${UNSEAL_PATH}
+./go-filecoin retrieval-client retrieve-piece "${MN_PEER_ID}" "${PIECE_1_CID}" \
+  --repodir="${CL_REPO_DIR}" > "${UNSEAL_PATH}"
 
 
 
-if [ "${USE_SMALL_SECTORS}" = true ] ; then
-    GOT=$(cat ${UNSEAL_PATH})
+GOT=$(shasum < "${UNSEAL_PATH}")
+EXPECTED=$(shasum < "${PIECE_1_PATH}")
 
-    if [ "${GOT}" = "${HODL}" ]; then
-        echo "Round trip passed!"
-        exit 0
-    else
-        echo "Round trip Failed!, expected:"
-        echo "${HODL}"
-        echo "got:"
-        echo "${GOT}"
-        exit 1
-    fi
+if [ "${GOT}" = "${EXPECTED}" ]; then
+    echo "Round trip passed!"
+    exit 0
 else
-    echo "big sector piece is too... big to cat"
-    echo ""
-    echo "report on unsealed file..."
-    du -sh ${UNSEAL_PATH}
+    echo "Round trip Failed!, expected file"
+    echo "${UNSEAL_PATH}"
+    echo "to have same contents as file"
+    echo "${PIECE_1_PATH}"
+    exit 1
 fi

